### PR TITLE
[3.0]Cleanup EventDispatcher mechanism

### PR DESCRIPTION
**Pull request by kevinw66**
## What is the purpose of the change

Cleanup EventDispatcher mechanism

### CODE REVIEW

File: dubbo-registry/dubbo-registry-kubernetes/src/main/java/org/apache/dubbo/registry/kubernetes/KubernetesServiceDiscovery.java [hunk 1]
```diff
@@ -194,21 +194,22 @@ public ServiceInstance getLocalInstance() {
 
     @Override
     public void addServiceInstancesChangedListener(ServiceInstancesChangedListener listener) throws NullPointerException, IllegalArgumentException {
+        super.addServiceInstancesChangedListener(listener);
         listener.getServiceNames().forEach(serviceName -> {
             SERVICE_UPDATE_TIME.put(serviceName, new AtomicLong(0L));
 
             // Watch Service Endpoint Modification
-            watchEndpoints(listener, serviceName);
+            watchEndpoints(serviceName);
```
Comment (chickenlj) [inline dubbo-registry/dubbo-registry-kubernetes/src/main/java/org/apache/dubbo/registry/kubernetes/KubernetesServiceDiscovery.java:202]: Does this change still match the original meaning?

The original impl only notify one listener while the new one notifies all listeners.

File: dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperServiceDiscoveryChangeWatcher.java [hunk 2]
```diff
@@ -68,7 +63,7 @@ public ZookeeperServiceDiscoveryChangeWatcher(ZookeeperServiceDiscovery zookeepe
         this.notifier = new RegistryNotifier(zookeeperServiceDiscovery.getDelay(), getExecutorRepository().getServiceDiscoveryAddressNotificationExecutor()) {
             @Override
             protected void doNotify(Object rawAddresses) {
-                listeners.forEach(listener -> listener.onEvent((ServiceInstancesChangedEvent)rawAddresses));
+                zookeeperServiceDiscovery.dispatchServiceInstancesChangedEvent((ServiceInstancesChangedEvent)rawAddresses);
```
Comment (chickenlj) [inline dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperServiceDiscoveryChangeWatcher.java:66]: Can listeners in Watcher being replaced with listeners in ServiceDiscovery?

Watcher is bound to one single service while ServiceDiscovery is shared among services.

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/support/AbortPolicyWithReport.java [hunk 3]
```diff
@@ -73,27 +71,18 @@ public AbortPolicyWithReport(String threadName, URL url) {
...
+                        " Thread Name: %s, Pool Size: %d (active: %d, core: %d, max: %d, largest: %d)," +
+                        " Task: %d (completed: %d)," +
+                        " Executor status:(isShutdown:%s, isTerminated:%s, isTerminating:%s), in %s://%s:%d!",
+                threadName, e.getPoolSize(), e.getActiveCount(), e.getCorePoolSize(), e.getMaximumPoolSize(),
+                e.getLargestPoolSize(),
+                e.getTaskCount(), e.getCompletedTaskCount(), e.isShutdown(), e.isTerminated(), e.isTerminating(),
+                url.getProtocol(), url.getIp(), url.getPort());
         logger.warn(msg);
         dumpJStack();
-        dispatchThreadPoolExhaustedEvent(msg);
```
[the remark this answers is not in the document]
Comment (kevinw66) [reply dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/support/AbortPolicyWithReport.java:85]: done

**Review by chickenlj — DISMISSED** [review]
LGTM.

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**